File: .agent/scripts/check_braces.py

import sys

def check_structure(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        lines = f.readlines()

    stack = []
    
    pairs = {')': '(', '}': '{', ']': '['}

    for i, line in enumerate(lines):
        line_num = i + 1
        for char in line:
            if char in '({[':
                stack.append((char, line_num))
            elif char in ')}]':
                if not stack:
                    print(f"Error: Unmatched '{char}' at line {line_num}")
                    return
                last_char, last_line = stack.pop()
                if last_char != pairs[char]:
                    print(f"Error: Mismatched '{char}' at line {line_num}. Expected closing for '{last_char}' from line {last_line}")
                    return

    if stack:
        char, line = stack[-1]
        print(f"Error: Unclosed '{char}' from line {line}")
    else:
        print("Structure looks OK")

if __name__ == "__main__":
    if len(sys.argv) < 2:
        print("Usage: python check_braces.py <filename>")
    else:
        check_structure(sys.argv[1])

File: .agent/scripts/check_braces.js

const fs = require('fs');

function checkStructure(filename) {
    try {
        const content = fs.readFileSync(filename, 'utf-8');
        const lines = content.split('\n');
        const stack = [];
        const pairs = { ')': '(', '}': '{', ']': '[' };
        
        let inString = false;
        let stringChar = '';

        for (let i = 0; i < lines.length; i++) {
            const line = lines[i];
            const lineNum = i + 1;
            
            for (let j = 0; j < line.length; j++) {
                const char = line[j];
                
                // Simple string skipping (doesn't handle escaped quotes perfectly but better than nothing)
                if (inString) {
                    if (char === stringChar && line[j-1] !== '\\') {
                        inString = false;
                    }
                    continue;
                }
                
                if (char === '"' || char === "'" || char === '`') {
                    inString = true;
                    stringChar = char;
                    continue;
                }

                if (['(', '{', '['].includes(char)) {
                    stack.push({ char, line: lineNum });
                } else if ([')', '}', ']'].includes(char)) {
                    if (stack.length === 0) {
                        console.log(`Error: Unmatched '${char}' at line ${lineNum}`);
                        return;
                    }
                    const last = stack.pop();
                    if (last.char !== pairs[char]) {
                        console.log(`Error: Mismatched '${char}' at line ${lineNum}. Expected closing for '${last.char}' from line ${last.line}`);
                        return;
                    }
                }
            }
        }

        if (stack.length > 0) {
            const last = stack[stack.length - 1];
            console.log(`Error: Unclosed '${last.char}' from line ${last.line}`);
        } else {
            console.log("Structure looks OK");
        }

    } catch (err) {
        console.error("Error reading file:", err);
    }
}

if (process.argv.length < 3) {
    console.log("Usage: node check_braces.js <filename>");
} else {
    checkStructure(process.argv[2]);
}
